WooCommerce/Classes/ViewRelated/Products/ProductsViewController.swift

```diff
@@ -1361,18 +1360,23 @@ extension ProductsViewController: SyncingCoordinatorDelegate {
         let action = AppSettingsAction.loadProductsSettings(siteID: siteID) { [weak self] (result) in
             switch result {
             case .success(let settings):
-                self?.syncProductCategoryFilterRemotely(from: settings) { settings in
+                self?.syncProductCategoryFilterRemotely(from: settings) { [weak self] settings in
+                    guard let self else { return }
                     if let sort = settings.sort {
-                        self?.sortOrder = ProductsSortOrder(rawValue: sort) ?? .default
+                        sortOrder = ProductsSortOrder(rawValue: sort) ?? .default
                     }
 
                     let promotableProductType = settings.productTypeFilter.map { PromotableProductType(productType: $0, isAvailable: true, promoteUrl: nil) }
-                    self?.filters = FilterProductListViewModel.Filters(stockStatus: settings.stockStatusFilter,
-                                                                       productStatus: settings.productStatusFilter,
-                                                                       promotableProductType: promotableProductType,
-                                                                       productCategory: settings.productCategoryFilter,
-                                                                       numberOfActiveFilters: settings.numberOfActiveFilters())
-
+                    filters = FilterProductListViewModel.Filters(stockStatus: settings.stockStatusFilter,
+                                                                 productStatus: settings.productStatusFilter,
+                                                                 promotableProductType: promotableProductType,
+                                                                 productCategory: settings.productCategoryFilter,
+                                                                 numberOfActiveFilters: settings.numberOfActiveFilters())
+
+                    // Emits `onDataReloaded` when local product settings (filters & sort order) are loaded and synced, so that
+                    // the first product selected in `selectFirstProductIfAvailable` is only triggered when the results match
+                    // the product settings.
+                    onDataReloaded.send(())
                 }
             case .failure:
                 break
```

Comment:
I see this was here from before this PR: If the sync result fails we just seem to attempt to re-synchronize through the `syncingCoordinator`, if we don't plan to do anything else with the `.failure` case, should we at least log it via `DDLogError`?

Reply:
Good point! Added a logging in https://github.com/woocommerce/woocommerce-ios/pull/12049/commits/2d94a7133a4e5338b037cc257a87329dfa88fbf5, and moved the `onDataReloaded` emission to a better place in the last two commits.